libsolidity/interface/SourceReferenceFormatter.cpp

```diff
@@ -60,19 +69,19 @@ void SourceReferenceFormatter::printSourceLocation(SourceLocation const* _locati
 			endColumn = startColumn + locationLength;
 		}
 
-		m_stream << line << endl;
+		m_stream << line.substr(0, startColumn);
+		m_stream << color << line.substr(startColumn, locationLength) << termcolor::reset;
+		m_stream << line.substr(endColumn) << endl;
 
 		for_each(
 			line.cbegin(),
 			line.cbegin() + startColumn,
 			[this](char const& ch) { m_stream << (ch == '\t' ? '\t' : ' '); }
 		);
-		m_stream << "^";
-		if (endColumn > startColumn + 2)
-			m_stream << string(endColumn - startColumn - 2, '-');
-		if (endColumn > startColumn + 1)
-			m_stream << "^";
-		m_stream << endl;
+
+		m_stream << termcolor::bold << color;
+		m_stream << string(locationLength, '^');
```

Comment:
I think we can split this single change out `^--^` to `^^^^`. Do we agree to make that change in the first place? I am quite indifferent, if other compilers prefer the `^^^^` option then I'd vote for going for that for easier onboarding.

Reply:
Yes, I realized that I had left it in after submitting the PR. A quick review of modern compiler error messages (Rust, Dotty (Scala), Elm) show a tendency towards ^^^^, while clang uses ^~~~ or similar. I find ^^^^ in general less distracting than ^---^.